crates/matrix-sdk/src/widget_api/handler/api.rs

```diff
@@ -4,23 +4,28 @@ use tokio::sync::oneshot::Receiver;
 use super::{
     super::messages::{capabilities::Options as CapabilitiesReq, openid},
     capabilities::Capabilities,
-    Outgoing, Result,
+    OutgoingMessage, Result,
 };
 
-#[async_trait(?Send)]
-pub trait Driver {
+#[async_trait]
+pub trait Client: Send + Sync + 'static {
     async fn initialise(&mut self, req: CapabilitiesReq) -> Result<Capabilities>;
-    async fn send(&self, message: Outgoing) -> Result<()>;
     async fn get_openid(&self, req: openid::Request) -> OpenIDState;
 }
 
+#[async_trait]
+pub trait Widget: Send + Sync + 'static {
+    async fn send<T: OutgoingMessage>(&self, message: T) -> Result<T::Response>;
+    fn early_init(&self) -> bool;
```

Comment:
we should call it: `init_on_load` this way we have the phrasing from the action in there (`Init on iframe load`) and also make it clear that we still wait for the widget to be loaded and not init before that.
```suggestion
    // If set to false the initialization is not done when the widget loaded, instead we wait until the ContentLoaded action is send
    fn init_on_load(&self) -> bool;
```

Reply:
The only issue that I have with "on load" wording is that from the perspective of the handler, it's not clear what "on load" means. For the handler, it's just a flag that says if the capabilities must be negotiated upon creation of the handler (and the handler is created once the widget/iframe is loaded).

Reply:
what if we swap the logic and call it initialize on content loaded

Reply:
This sounds like a great idea! Done.